BootloaderCorePkg/Tools/BuildUtility.py

```diff
@@ -548,24 +538,31 @@ def gen_payload_bin (fv_dir, pld_list, pld_bin, priv_key, brd_name = None):
     # E-payloads container format
     alignment = 0x10
     key_dir  = os.path.dirname (priv_key)
-    pld_list = [('EPLD', '%s' % epld_bin, '', 'RSA2048', '%s' % os.path.basename(priv_key), alignment, 0)]
+    key_type = get_key_type(priv_key)
+    pld_list = [('EPLD', '%s' % epld_bin, '', key_type, '%s' % os.path.basename(priv_key), alignment, 0)]
     for pld in ext_list:
-        pld_list.append ((pld['name'], pld['file'], pld['algo'], 'SHA2_256', '', 0, 0))
+        pld_list.append ((pld['name'], pld['file'], pld['algo'], hash_alg, '', 0, 0))
     gen_container_bin ([pld_list], fv_dir, fv_dir, key_dir, '')
 
+def pub_key_valid (pubkey):
+    if len(pubkey) != (0x104 + sizeof(PUB_KEY_HDR)) and len(pubkey) != (0x184 + sizeof(PUB_KEY_HDR)):

```

Comment:
Use  (len(pubkey) - sizeof(PUB_KEY_HDR)) in [0x104, 0x184]:
And it is better to define key length as const for easier read.

Reply:
Thanks. I would improve this method.